apps/os/backend/agent/do-tools.ts

```diff
@@ -88,3 +91,35 @@ export function doToolToRuntimeJsonSchema(_value: unknown) {
   };
   return runtimeJsonSchema;
 }
+
+export function createDOToolFactory<TAgent extends SlackAgent | IterateAgent>() {
```

Comment:
This function shouldn't know about SlackAgent | IterateAgent

Reply:
Discussed in person - it's extending these to enable cmd-click into the method implementations itself. Let's cmd-click into the tool defs instead